FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -211,57 +221,52 @@ extension ComposeViewController {
-            self?.showAlert(error: error, message: "compose_error".localized)
-        }
+        encryptAndSendMessage()
     }
 
-    private func encryptAndSendMessage() -> Promise<Bool> {
-        Promise<Bool> { [weak self] () -> Bool in
-            guard let self = self else { return false }
-
-            let recipients = self.contextToSend.recipients
-
-            guard recipients.isNotEmpty else {
-                assertionFailure("Recipients should not be empty. Fail in checking")
-                return false
-            }
-
-            guard let text = self.contextToSend.message else {
-                assertionFailure("Text and Email should not be nil at this point. Fail in checking")
-                return false
-            }
-
-            let subject = self.input.subjectReplyTitle
-                ?? self.contextToSend.subject
-                ?? "(no subject)"
-
-            guard let myPubKey = self.dataService.publicKey else {
-                self.showAlert(message: "compose_no_pub_sender".localized)
-                return false
-            }
-
-            guard let allRecipientPubs = self.getPubKeys(for: recipients) else {
-                return false
-            }
-
-            let encrypted = self.encryptMsg(
-                pubkeys: allRecipientPubs + [myPubKey],
-                subject: subject,
-                message: text,
-                to: recipients.map { $0.email }
-            )
-
-            try await(self.messageSender.sendMail(mime: encrypted.mimeEncoded))
-
-            return true
+    private func encryptAndSendMessage() {
+        let recipients = self.contextToSend.recipients
+        
+        guard recipients.isNotEmpty else {
+            return
+        }
+        
+        guard let text = self.contextToSend.message else {
+            return
+        }
+        
+        let subject = self.input.subjectReplyTitle
+            ?? self.contextToSend.subject
+            ?? "(no subject)"
+        
+        guard let myPubKey = self.dataService.publicKey else {
+            return
         }
+        
+        guard let allRecipientPubs = getPubKeys(for: recipients) else {
+            return
+        }
+        
+        let encrypted = self.encryptMsg(
+            pubkeys: allRecipientPubs + [myPubKey],
+            subject: subject,
+            message: text,
+            to: recipients.map { $0.email }
+        )
+            
+        self.messageSender
+            .sendMail(mime: encrypted.mimeEncoded)
+            .sink(
+                receiveCompletion: { [weak self] result in
+                    guard case .failure(let error) = result else {
+                        return
+                    }
+                    self?.showAlert(error: error, message: "compose_error".localized)
+                },
+                receiveValue: {  [weak self] in
+                    self?.handleSuccessfullySentMessage()
+                })
+            .store(in: &cancellable)
```

Comment:
This PR looks good, and we could actually clean up and merge it.

What does the `.sink` represent?

I figured `receiveCompletion` triggers on both error and success, and you are filtering for error only? Isn't there a more convenient method like `receiveFailure`?

What does the `.store(in: &cancellable)` do?

Reply:
sink is used to "subscribe" to stream of events.
receive completion will be triggered in case of error and in case when stream is finished
```
 .sink(
   receiveCompletion: { completion in
      switch completion {
      case .failure(let error): print("Error \(error)")
      case .finished: print("Publisher is finished")
      }
   }
   ...
```
In this example with Future(promise), there will be only 1 value, which can be accessed in `receiveValue:` closure and 1 receiveCompletion. 
But in cases with other publishers, receiveValue closure can be called multiple times but `receiveCompletion` - will be called only once when publisher finished or failed
 
There is `sink(receiveValue:)` function, but there is no just `receiveError`.

`.store(in: &cancellable)`. 
In case we started some long going request and left the screen cancellable will frees up any allocated resources and also will stop side effects. 
For example user start fetching message with bad internet connection and was waiting for few seconds. Then taped back and left the screen, cancelable will terminate this execution and free up resources.

Reply:
All subscriptions should be stored in cancellable to prevent their execution if view controller/service is deallocated